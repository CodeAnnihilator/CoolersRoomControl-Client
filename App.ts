import App from './src/root/App';

export default App;
