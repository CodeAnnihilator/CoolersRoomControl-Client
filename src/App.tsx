import React from 'react';
import {createAppContainer} from 'react-navigation';

import AppNavigator from './navigation/components/index';

const AppContainer = createAppContainer(AppNavigator);

export default class App extends React.Component {
	public render() {
		return <AppContainer />;
	}
}
