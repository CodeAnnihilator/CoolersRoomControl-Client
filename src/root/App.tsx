import {Font} from 'expo';
import * as React from 'react';
import {Provider} from 'react-redux';
import {createAppContainer} from 'react-navigation';
import EStyleSheet from 'react-native-extended-stylesheet';

import COLORS from '../common/constants/colors';

import configureStore from './configureStore';

import AppNavigator from '../navigation/components/index';

EStyleSheet.build(COLORS);

const store = configureStore({});

const AppContainer = createAppContainer(AppNavigator);

interface IAppState {
	fontLoaded: boolean;
}

export const asyncLoadingFonts = async () => {
	await Font.loadAsync({
		'DIN': require('../assets/fonts/D-DIN.ttf'),
		'DIN-bold': require('../assets/fonts/D-DIN-Bold.ttf'),
	});
};

export default class App extends React.Component<{}, IAppState> {
	public state = {
		fontLoaded: false,
	};

	public async componentDidMount() {
		await asyncLoadingFonts();
		this.setState({fontLoaded: true});
	}

	public render() {
		return (
			<Provider store={store}>
				{this.state.fontLoaded ? <AppContainer /> : null}
			</Provider>
		);
	}
}
