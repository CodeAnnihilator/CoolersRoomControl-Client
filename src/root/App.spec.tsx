import * as React from 'react';
import {create} from 'react-test-renderer';

import App, {asyncLoadingFonts} from './App';

describe('App', () => {
	it('renders with async fonts being loaded', async () => {
		try {
			await asyncLoadingFonts(); // TODO: test w/wo fonts
			const tree = create(<App />).toJSON();

			return expect(tree).toMatchSnapshot();
		} catch (e) {
			expect(e).toMatch('error');
		}
	});
});
