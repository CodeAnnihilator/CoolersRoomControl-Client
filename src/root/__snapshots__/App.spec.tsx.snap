// Jest Snapshot v1, https://goo.gl/fbAQLP

exports['App renders with async fonts being loaded 1'] = 'null';
